src/main/java/com/google/sps/data/EventUpdater.java

```diff
@@ -36,7 +53,163 @@ public Entity getEntity() {
+    }
+
+    HashMap<String, String> formProperties = new HashMap<String, String>();
+
+    // Format is (Form EntryName, Entity PropertyName)
+    formProperties.put("event-primary-organization-id", "eventOwnerOrgId");
+    formProperties.put("event-name", "eventName");
+    formProperties.put("event-partner", "eventPartnerNames");
+    formProperties.put("event-details", "eventDetails");
+    formProperties.put("event-contact-email", "eventContactEmail");
+    formProperties.put("event-contact-phone-num", "eventContactPhone");
+    formProperties.put("event-contact-name", "eventContactName");
+    formProperties.put("event-street-address", "eventStreetAddress");
+    formProperties.put("event-city", "eventCity");
+    formProperties.put("event-state", "eventState");
+    formProperties.put("event-zip-code", "eventZipcode");
+
+    /* Optional Properties can be left blank in the request form.
+     *
+     * The following properties are optional:
+     * - eventPartnerNames
+     * - eventDetails
+     */
+    Set<String> optionalProperties = new HashSet<String>();
+    optionalProperties.add("eventPartnerNames");
+    optionalProperties.add("eventDetails");
+
+    for (Map.Entry<String, String> entry: formProperties.entrySet()) {
+      String propertyKey = entry.getValue();
+      String formKey = entry.getKey();
+      String formValue = "";
+
+      if (formKey.equals("event-primary-organization-id")) {
+        continue;
+      }
+
+      if (optionalProperties.contains(propertyKey)) {
+        formValue = request.getParameter(formKey) == null ? "" : request.getParameter(formKey);
+      } else {
+        try {
+          formValue = RequestHandler.getParameterOrThrow(request, formKey);
+        } catch (IllegalArgumentException err) {
+          logger.log(Level.SEVERE, "Form value for: " + propertyKey + " cannot be left blank.");
+          throw new IllegalArgumentException();
+        }
+      }
+
+      setEventProperty(propertyKey, formValue);
+    }
+
+    setNonFormProperties(forRegistration, historyUpdate);
+
+    setEventDateAndHours(request);
+
+    setEventOwnerOrgIdAndName(ownerOrgId);
+  }
+
+  private Entity getOrgEntityWithId(long orgId) throws IllegalArgumentException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Key organizationKey = KeyFactory.createKey("Distributor", orgId);
+
+    Entity organizationEntity = null;
+    try {
+      organizationEntity = datastore.get(organizationKey);
+    } catch (com.google.appengine.api.datastore.EntityNotFoundException err) {
+      throw new IllegalArgumentException("Organization entity with orgID " + orgId + " was not found.");
+    }
+
+    if (organizationEntity == null) {
+      throw new IllegalArgumentException("There is no Organization with ID: " + orgId);
+    }
+
+    return organizationEntity;
+  }
+
+  private boolean doesUserHasCredentialsToUpdateEvent(GivrUser user, long orgId) {
+    Entity entity = getOrgEntityWithId(orgId);
```

Comment:
I thought there was a 
GiverUser.isModeratorOfOrg(long orgId) {} function. If there isn't can you make one?

Reply:
Added! I moved getOrgEntityWithId into Organization class since it doesn't deal with updating Organization. Is this okay?